app/liquid_tags/github_tag/github_readme_tag.rb

```diff
@@ -4,38 +4,35 @@ class GithubTag
   class GithubReadmeTag
     PARTIAL = "liquids/github_readme".freeze
 
+    attr_reader :client, :content, :options, :readme_html
+
     def initialize(link)
-      @link = parse_link(link)
+      parsed_link = parse_link(link)
```

Comment:
IMHO a better name. Also, this doesn't need to be an instance variable since it gets passed into all methods relying on it.

I also moved link validation + raising of an error into `parse_link`. Before we parsed the link, then validated it in `get_content`. Now we exit as early as possible, avoiding all unncessary work.